### ISSUE TITLE
Unable to merge upstream changes

### ISSUE THREAD

**Issue by fopina** [surface-security/httpx#19]
### Related software versions

*

### Description

As the history was abrutly reset when the code was made public, the tree is no longer shared with upstream httpx repository making it harder to get the updates.

I suppose we could script cherrypicks from upstream but it's easier to cherrypick once now to get the history properly done!

I have already done in the branch `temporary_redo`.

I pulled upstream tag `v1.1.2` into an orphan branch and then I cherry picked our `Initial commit` and resolved all the conflicts (very few).

I'll now cherry pick the rest of the history up until now.

@gsilvapt @mariandppb if you guys are ok with it, we can just delete `main` and rename `temporary_redo` to `main`. From then on, merging upstream is possible again.

### Use cases

Keep httpx up to date

**Comment by fopina** [comment]
No idea how I managed to screw up by opening this issue here, but this is related to httpx as mentioned in the description...

https://github.com/surface-security/scanner-httpx/tree/temporary_redo

**Comment by gsilvapt** [comment]
> No idea how I managed to screw up by opening this issue here, but this is related to httpx as mentioned in the description...
> 
> https://github.com/surface-security/scanner-httpx/tree/temporary_redo

We can transfer issues from repositories, on the right side of the issue page. I've move it to the correct place regardless :)

---


This explains why we were having so many issues trying to update the fork :sweat_smile: 

Since ours is relatively old, do you think we still need the fork? As in, now the scanner supports users to specify headers and we could inject there the headers we want. I'm not totally sure what else justified the fork, maybe we cannot drop it but it's worth reviewing at this point.

If we somehow could just use `httpx` directly that would be great (and would be one less thing to maintain) :ok_hand:

**Comment by fopina** [comment]
I think it’s easy to move this to an entrypoint like most of the other scanners, but the main  change is the added output directory which would have to be done in entrypoint (same as the other scanners).

Not sure if it makes any easier to update though. With a clean tree like this and little changes (of ours) moving forward, there should be very few conflicts when we want to merge upstream.

Using the entrypoint still has a burden to maintain as the CLI flags might change across versions.
But I don’t have anything against if that’s the preference

**Comment by gsilvapt** [comment]
I'm just trying to evaluate what is the option that leads to less maintenance overall. If these changes leads to less of those, then I guess it's fine to keep it?

**Comment by fopina** [comment]
Yes, I’d vote we can try to keep it as it was originally (not the broken tree) and see how it goes 👌

**Comment by fopina** [comment]
should we take the opportunity to reset/cleanup latest commit history? it looks a bit "testish" (which could have been done in a non-main branch and merged only after done)

**Comment by gsilvapt** [comment]
Yea, sounds like a good idea to me :ok_hand:

**Comment by fopina** [comment]
`dev-112-tree-cleanup` is the new main branch then

https://github.com/surface-security/scanner-httpx/actions/runs/4114083379 should validate that build is working after squashing 🙏 

I no longer have permissions to modify protected branches, feel free to proceed @gsilvapt 😄

**Comment by gsilvapt** [comment]
`dev-112-tree-cleanup` is now the current main. Is it safe to delete the old `main` (now `main-2`)? Just to avoid conflicts and/or unexpected behavior.

**Comment by fopina** [comment]
I think it's safe yes, but if you prefer to keep it, we could use some weirder name to keep it away from search by main branch, like `old-broken-tree-main-do-not-use`.

I'll this issue then (but re-open at will if I should not)
